api/src/app/modules/submission/submission.service.ts

```diff
@@ -664,15 +667,66 @@ export class SubmissionService extends DataService<Submission, Repository<Submis
       await this.updateProjectLocation(submission.projectId, user); // This will set geometry_latlong to null.
     }
     else {
-      await this.saveAndUpdateSpatialSubmission(submission, spatialObjectCode, user);
+      await this.repository.save(submission);
+      await this.updateProjectLocationAndSpatialAreaOrLength(submission, spatialObjectCode, user)
```

Comment:
This is when removing one spatial type object from the submission, then we save the submission and update project location and area/length. No need to do update on geometry simplify.